### PR TITLE
Trim Chunks if LLM tokenizer differs from Embedding tokenizer

### PR DESCRIPTION
**Pull request by yuhongsun96**

### CODE REVIEW

File: backend/danswer/chat/chat_utils.py [hunk 1]
```diff
@@ -296,8 +305,19 @@ def get_chunks_for_qa(
 
             # We calculate it live in case the user uses a different LLM + tokenizer
             chunk_token = check_number_of_tokens(chunk.content)
+            if chunk_token > DOC_EMBEDDING_CONTEXT_SIZE + token_leeway:
```
Comment (yuhongsun96) [inline backend/danswer/chat/chat_utils.py:308]: UCSD related issue change

File: backend/danswer/configs/model_configs.py [hunk 2]
```diff
@@ -3,7 +3,6 @@
 #####
 # Embedding/Reranking Model Configs
 #####
-CHUNK_SIZE = 512
```
Comment (yuhongsun96) [inline backend/danswer/configs/model_configs.py:6]: Removed duplicate variable used for the same purpose

File: backend/danswer/main.py [hunk 3]
```diff
@@ -130,8 +132,124 @@ def include_router_with_global_prefix_prepended(
     application.include_router(router, **final_kwargs)
 
 
+@asynccontextmanager
+async def lifespan(app: FastAPI) -> AsyncGenerator:
```
Comment (yuhongsun96) [inline backend/danswer/main.py:136]: Removed deprecated on_startup function for this

File: backend/danswer/search/search_nlp_models.py [hunk 4]
```diff
@@ -26,10 +26,10 @@
 
 
 os.environ["TOKENIZERS_PARALLELISM"] = "false"
+os.environ["HF_HUB_DISABLE_TELEMETRY"] = "1"
```
Comment (yuhongsun96) [inline backend/danswer/search/search_nlp_models.py:29]: should have done this a long time ago

File: backend/danswer/search/search_nlp_models.py [hunk 5]
```diff
@@ -26,10 +26,10 @@
...
 
 logger = setup_logger()
-# Remove useless info about layer initialization
-logging.getLogger("transformers").setLevel(logging.ERROR)
+transformer_logging.set_verbosity_error()
```
Comment (yuhongsun96) [inline backend/danswer/search/search_nlp_models.py:32]: This way actually works, the other did not

File: backend/danswer/utils/timing.py [hunk 6]
```diff
@@ -18,7 +18,9 @@
 
 
 def log_function_time(
-    func_name: str | None = None, print_only: bool = False
+    func_name: str | None = None,
+    print_only: bool = False,
+    debug_only: bool = False,
```
Comment (yuhongsun96) [inline backend/danswer/utils/timing.py:23]: ended up not using this change but its fine to have this flexibility

File: backend/danswer/chat/chat_utils.py (hunk 1 above)
Comment (Weves) [reply backend/danswer/chat/chat_utils.py:308]: Shouldn't we only do this if `not latest_batch_indices`?

Comment (Weves) [reply backend/danswer/chat/chat_utils.py:308]: nvm discussed this way is fine

**Pull request merged**